app/services/factories/course_identifier.rb

```diff
@@ -10,7 +10,7 @@ def call(course)
     private
 
       def recorder_klass_name_for_course_identifier(course)
-        declaration_identifiers[course.underscore.intern].to_s
+        declaration_identifiers[course.underscore.intern].to_s if course
```

Comment:
Is this enough. Should we also have the caller with a catch block as per the event factory?

Reply:
```
      def call(course)
        recorder_klass_name_for_course_identifier(course).presence || (raise ActionController::ParameterMissing, I18n.t(:invalid_course))
      end
```
should prevent the caller from doing anything.

Reply:
Alright, I made it similar to event factory :)